### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by leewan-09**
### Changes proposed

- Improved avatar component 
- improved ESlint config
- remove the unwanted package
- types improved

### CODE REVIEW

**Comment by rohitdasu** [comment]
@leewan-09 could you please check the build issue

**Comment by leewan-09** [comment]
why i cant view

**Comment by rohitdasu** [comment]
@leewan-09 i believe it's because you don't have access to the vercel. but you can try to build at your local machine and debug. that would be same.

**Comment by rohitdasu** [comment]
@leewan-09 i think there's an issue occurred in modal component. please could you check it

Screenshot 2022-09-10 at 7 54 38 PM

**Comment by leewan-09** [comment]
Checking

File: components/AuthModal/AuthModal.tsx [hunk 1]
```diff
@@ -6,10 +6,9 @@ import { closeModal } from '../../store/slices/sliceModal';
 import { signInWithPopup } from 'firebase/auth';
 import { auth, Provider_github, Provider_google } from '../../lib/firebase';
 import toast, { Toaster } from 'react-hot-toast';
-import tw from 'twin.macro';
```
Comment (rohitdasu) [inline components/AuthModal/AuthModal.tsx:9]: i think this caused the issue of modal

File: components/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -8,7 +8,12 @@ import { useDispatch } from 'react-redux';
 import { setUserLoggedOut } from '../../store/slices/sliceUser';
 import toast from 'react-hot-toast';
 
-export function Avatar({ userImg }: any) {
+type AvatarProps = {
+  userImg: string | null | undefined;
+  email: string | null | undefined;
+};
```
Comment (rohitdasu) [inline components/Avatar/Avatar.tsx:14]: can we keep it inside avatar.interface.ts?

**Review by rohitdasu — CHANGES_REQUESTED** [review]
@leewan-09 could you put interfaces/types to their respective component's  <component>.interface.ts file?

File: pages/_document.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import Document, { Html, Head, Main, NextScript } from 'next/document';
 import { extractCritical } from '@emotion/server';
 
 export default class MyDocument extends Document {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (rohitdasu) [inline pages/_document.tsx:6]: could you follow this for fixing the eslint error
https://stackoverflow.com/questions/67087999/how-to-properly-type-the-document-tsx-file-from-next-js

File: components/AuthModal/AuthModal.tsx [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
+import 'twin.macro';
 
 type Auth_data = {
-  title: String;
+  title: string;
 };
```
Comment (rohitdasu) [inline components/AuthModal/AuthModal.tsx:12]: this can be put inside the AuthModal.interface.ts

File: components/Avatar/avatar.interface.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export type IAvatar = {
+  userImg: string | null | undefined;
+  email: string | null | undefined;
+};
```
Comment (rohitdasu) [inline components/Avatar/avatar.interface.ts:4]: filename should be Avatar.interface.ts

Comment (rohitdasu) [reply components/Avatar/avatar.interface.ts:4]: also please make sure you add an index.ts file in the component and exporting interface and component as well. please check other components for reference

File: components/AuthModal/AuthModal.tsx (hunk 4 above)
Comment (rohitdasu) [reply components/AuthModal/AuthModal.tsx:12]: naming convention is also not correct. it should be AuthData

**Comment by leewan-09** [comment]
Why did the building fail?  I have tested it locally; it's fine.

**Comment by rohitdasu** [comment]
this is the error which i can see in vercel terminal @leewan-09 

Screenshot 2022-09-10 at 9 00 30 PM

File: components/Avatar/avatar.interface.ts (hunk 5 above)
Comment (rohitdasu) [reply components/Avatar/avatar.interface.ts:4]: Please change this file's first letter "Capital"

File: components/Avatar/Avatar.tsx [hunk 6]
```diff
@@ -7,8 +7,9 @@ import { useAppSelector } from '../../app/hooks';
 import { useDispatch } from 'react-redux';
 import { setUserLoggedOut } from '../../store/slices/sliceUser';
 import toast from 'react-hot-toast';
+import { IAvatar } from './index';
```
Comment (rohitdasu) [inline components/Avatar/Avatar.tsx:10]: i think it should be imported from the interface file not from the index here

**Comment by rohitdasu** [comment]
you need to make the filename change. please see. 

Screenshot 2022-09-10 at 9 07 14 PM

**Comment by rohitdasu** [comment]
> you need to make the filename change. please see.
> 
> Screenshot 2022-09-10 at 9 07 14 PM

you need to make it "Avatar.interface.ts".

**Comment by leewan-09** [comment]
Look I have changed like that..

image

**Comment by rohitdasu** [comment]
it's still showing the lower case in github. strange

**Comment by leewan-09** [comment]
Yes, very strange.

**Comment by leewan-09** [comment]
image

Look in here also

**Comment by rohitdasu** [comment]
@leewan-09 could you delete that interface file and push it. after that again create that file and push. let's see if that updates in github

**Comment by leewan-09** [comment]
Ahhh. now it's okay, That was strange

**Review by rohitdasu — APPROVED** [review]
LGTM Now 👍

**Pull request merged**